src/core/dom/serialized-event-listener.js

```diff
@@ -0,0 +1,42 @@
+import * as mode from '#core/mode';
+
+let shouldSerialize = false;
+
+/**
+ * In any mode except test, this function behaves calls addEventListener and exits.
+ * In test mode, it provides the ability to toggle "serialization" mode on, which
+ * serializes the event listener as attributes on the node.
+ *
+ * @param {Element} element
+ * @param {string} event
+ * @param {function(T):void} handler
+ * @template {Event} T
+ */
+export function addSerializedEventListener(element, event, handler) {
```

Comment:
This is only needed for compiler code/buildDom code right? How dirty would it be to do something like
```javascript
const originalAddEventListener = Element.prototype.addEventListener;
// define addSerializedEventListener with something like
/// `originalAddEventListener.call(element, event, handler)`
Element.prototype.addEventListener = addSerializedEventListener;
```

Taking over the prototype would let AMP/Bento devs write code without making compiler-specific changes

Reply:
Hmm. Temporarily stubbing within the test....that might be much cleaner. If possible

Reply:
Maybe ping Raghu, he might know if/where we do similar hackery for testing, especially with some of our integration/realWin etc tests